apps/web/src/components/views/CareerView.tsx

```diff
@@ -1,40 +1,118 @@
-import { type FC } from "react"
-import { type CareerProps } from "@/pages/career"
+import { useState, type FC } from "react"
 import CompanyAdListItem from "../molecules/CompanyAdListItem"
+import CompanyFiltersContainer from "../molecules/CompanyFiltersContainer"
+import { type JobListing } from "@dotkomonline/types"
+import OnlineIcon from "../atoms/OnlineIcon"
 
-const CareerView: FC<CareerProps> = (props: CareerProps) => (
+interface CareerProps {
+  careers: JobListing[]
+}
+
+export interface EmploymentCheckbox {
+  name: string
+  checked: Boolean
+}
+
+const CareerView: FC<CareerProps> = (props: CareerProps) => {
   // return <div> 404 Siden finnes ikke </div>
 
-  <div>
-    <div className="bg-amber-9 absolute left-0 top-[56px] h-[250px] w-full opacity-30" />
-    <div className="absolute left-0 top-[56px] flex h-[250px] w-full flex-col justify-center">
-      <div className="text-slate-3 m-auto h-[200px] max-w-[800px] p-5 text-center">
-        <p className="leading-1.4 mt-5  text-4xl font-bold">
-          Er du på jakt etter <span className="bg-amber-6 bg-center bg-no-repeat">jobb</span>?
-        </p>
-        <p className="leading-1.4 mt-7 text-2xl font-bold">
-          Her har du en liste over bedrifter som er ute etter deg som informatikkstudent!
-        </p>
+  const [chosenLocation, setChosenLocation] = useState<string>("Alle")
+  const [searchName, setSearchName] = useState<string>("")
+  const [chosenEmployments, setChosenEmployments] = useState<EmploymentCheckbox[]>([
+    { name: "Deltid", checked: false },
+    { name: "Fulltid", checked: false },
+    { name: "Sommerjobb/internship", checked: false },
+  ])
+  const [chosenSort, setChosenSort] = useState<string>("Frist")
+
+  function filterLocation(jobListing: JobListing) {
```

Comment:
I suggest either moving the functions out of the component or into a seperate file. This way the component is easier to read. It would also be nice with some simple tests for the functions with the logic